atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -62,25 +62,32 @@ public synchronized void shutdown() {
         log.info("Shutdown completed!");
     }
 
-    public synchronized void acquirePersistentLockWithRetry() {
+    // We don't synchronize this method, to avoid deadlocking if {@link #shutdown} is called while attempting
+    // to acquire a lock.
+    public void acquirePersistentLockWithRetry() {
+        while (!tryAcquirePersistentLock()) {
+            waitForRetry();
+        }
+    }
+
+    private synchronized boolean tryAcquirePersistentLock() {
         if (isShutDown) {
```

Comment:
It's not necessary for thread-safety, since `isShutDown` is only used in `synchronized` methods. However, it would provide additional protection, in case we use the variable in a non-`synchronized` method in future.